src/ucs/datastruct/arbiter.c

```diff
@@ -294,12 +296,16 @@ void ucs_arbiter_dispatch_nonempty(ucs_arbiter_t *arbiter, unsigned per_group,
                 } else {
                     /* Not only element */
                     ucs_assert(elem == last_elem->next); /* first element should be removed */
-                    if (group_head == prev_group) {
+                    if (is_single_group) {
                         next_group = next_elem; /* No more groups, point arbiter
                                                    to next element in this group */
                         ucs_list_head_init(&next_elem->list);
                     } else {
                         /* Insert the next element to the arbiter list */
+                        /* coverity reports issue on second iteration of inner loop,
+                         * it assumes that group_head or prev_group are changes during
+                         * loop iteration which is incorrect. just suppress error */
+                        /* coverity[deref_after_free] */
```

Comment:
maybe we can remove this coverity warning, now that we use is_single_group?

Reply:
removed

Reply:
coverity is passing now with latest version, right?

Reply:
yes, checked it